api/scpca_portal/management/commands/generate_computed_files.py

```diff
@@ -0,0 +1,146 @@
+
+    The directory structure for this bucket should follow this pattern:
+        /project_metadata.csv
+        /SCPCP000001/libraries_metadata.csv
+        /SCPCP000001/samples_metadata.csv
+        /SCPCP000001/SCPCS000109/SCPCL000126_filtered.rds
+        /SCPCP000001/SCPCS000109/SCPCL000126_metadata.json
+        /SCPCP000001/SCPCS000109/SCPCL000126_processed.rds
+        /SCPCP000001/SCPCS000109/SCPCL000126_qc.html
+        /SCPCP000001/SCPCS000109/SCPCL000126_unfiltered.rds
+        /SCPCP000001/SCPCS000109/SCPCL000127_filtered.rds
+        /SCPCP000001/SCPCS000109/SCPCL000127_metadata.json
+        /SCPCP000001/SCPCS000109/SCPCL000127_processed.rds
+        /SCPCP000001/SCPCS000109/SCPCL000127_qc.html
+        /SCPCP000001/SCPCS000109/SCPCL000127_unfiltered.rds
+
+    The files will be zipped up and stats will be calculated for them.
+
+    If run locally the zipped ComputedFiles will be copied to the
+    "scpca-local-data" bucket.
+
+    If run in the cloud the zipped ComputedFiles files will be copied
+    to a stack-specific S3 bucket."""
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--clean-up-input-data",
+            action=BooleanOptionalAction,
+            type=bool,
+            default=settings.PRODUCTION,
+        )
+        parser.add_argument(
+            "--clean-up-output-data",
+            action=BooleanOptionalAction,
+            type=bool,
+            default=settings.PRODUCTION,
+        )
+        parser.add_argument("--max-workers", type=int, default=10)
+        parser.add_argument("--scpca-project-id", type=str)
+        parser.add_argument(
+            "--update-s3", action=BooleanOptionalAction, type=bool, default=settings.UPDATE_S3_DATA
+        )
+
+    def handle(self, *args, **kwargs):
+        self.generate_computed_files(**kwargs)
+
+    @staticmethod
+    def clean_up_input_data() -> None:
+        shutil.rmtree(common.INPUT_DATA_PATH, ignore_errors=True)
+
+    @staticmethod
+    def clean_up_output_data() -> None:
+        shutil.rmtree(common.OUTPUT_DATA_PATH, ignore_errors=True)
+
+    def create_computed_file(self, future, *, update_s3: bool, clean_up_output_data: bool) -> None:
+        """
+        Saves computed file returned from future to the db.
+        Uploads file to s3 and cleans up output data depending on passed options.
+        """
+        if computed_file := future.result():
+
+            # Only upload and clean up projects and the last sample if multiplexed
+            if computed_file.project or computed_file.sample.is_last_multiplexed_sample:
+                if update_s3:
+                    s3.upload_output_file(computed_file.s3_key, computed_file.s3_bucket)
+                if clean_up_output_data:
+                    computed_file.clean_up_local_computed_file()
+            computed_file.save()
+
+        # Close DB connection for each thread.
+        connection.close()
+
+    def generate_computed_files(self, **kwargs) -> None:
```

Comment:
Same feedback as before where we want to have the arguments in the function definition.